edith-step/src/components/ReceiptHandler.js

```diff
@@ -0,0 +1,307 @@
+    };
+
+    /**
+     * Handles changes to the grocery store of the trip.
+     * @param {Event} e change event
+     */
+    this.handleStoreChange = (e) => {
+      this.setState({storeName: e.target.value});
+    };
+
+    /**
+     * Adds a form row for the user to input additional grocery items
+     * that weren't included in the initial scan.
+     * @param {Event} e change event
+     */
+    this.addItem = (e) => {
+      const newItem = {
+        userId: this.state.userId,
+        name: '',
+        price: 0.0,
+        quantity: 0,
+        category: '',
+        expiration: '',
+      };
+      this.setState({items: this.state.items.concat(newItem)});
+    };
+
+    /**
+     * Handles primary submission to the form. Sends the grocery
+     * data to the receipt-data servlet to find deals and expiration
+     * dates.
+     *
+     * @param {Event} e change event
+     */
+    this.handleSubmit = async (e) => {
+      e.preventDefault();
+      let price = 0;
+      this.state.items.forEach((item) => {
+        price += item.price * item.quantity;
+      });
+      this.setState({totalPrice: price});
+      const response = await axios({
+        method: 'post',
+        url: '/receipt-data',
+        data: {
+          userId: this.state.userId,
+          storeName: this.state.storeName,
+          date: this.state.date,
+          name: this.state.name,
+          fileUrl: this.state.fileUrl,
+          totalPrice: price,
+          items: this.state.items,
+        },
+      });
+
+      const data = response.data;
+      this.setState((state) => ({
+        deals: data,
+      }));
+    };
+
+    /**
+     * Handles secondary submission to the form. Sends the data
+     * to the store-receipt servlet to be stored in Datastore.
+     * @param {Event} e change event
+     */
+    this.addExpirationAndSubmit = async (e) => {
+      let price = 0;
+      this.state.items.forEach((item) => {
+        price += item.price * item.quantity;
+      });
+      this.setState({totalPrice: price});
+      await axios({
+        method: 'post',
+        url: '/store-receipt',
+        data: {
+          userId: this.state.userId,
+          storeName: this.state.storeName,
+          date: this.state.date,
+          name: this.state.name,
+          fileUrl: this.state.fileUrl,
+          totalPrice: price,
+          items: this.state.items,
+          deals: this.state.deals,
+        },
+      });
+      axios({
+        method: 'post',
+        url: '/user-stats-servlet',
+        data: {
+          itemName: this.state.itemName,
+          itemCategory: this.state.itemCategory,
+          itemPrice: this.state.itemPrice,
+          itemQuantity: this.state.itemQuantity,
+          itemDate: this.getDate(),
+          itemReceiptId: this.state.itemReceiptId,
+          itemCategory: this.state.itemCategory,
+        },
+      }).catch((err) => {
+        console.log(err);
+      });
+      this.setState({});
```

Comment:
I don't think this will change the state. Is this to force a re-render of the component?